src/pages/dataset/[dataset_id].js

```diff
@@ -30,7 +30,11 @@ export const Dataset = ({ query }) => {
   const pageRendered = usePageRendered()
   const { setResponsive } = useResponsive()
   const [selectedDataset, setSelectedDataset] = useState({})
-  const isSharedDataset = ref === 'share' || !selectedDataset.is_processed
+  const isSharedDataset =
+    (ref === 'share' &&
+      !selectedDataset.is_processed &&
+      selectedDataset.success !== false) ||
+    (!selectedDataset.is_processed && selectedDataset.success !== false)
```

Comment:
I'm unsure what this check is trying to accomplish. I'm going to change the names to illustrate the logic a little bit more clearly.
```js
const A = ref === 'share'
const B = selectedDataset.is_processed
const C = selectedDataset.success !== false
const isSharedDataset = (A && !B && C) || (!B && C)

```
which would functionally be the same as:
```js
const isSharedDataset = !B && C
```

So I think either we need to change the name and not check the share ref or reconsider this check.

Reply:
Now it checks the unprocessed status rather than the `share` ref. As a quick cleanup, I took this opportunity to also delete the shared dataset check from the `DatasetPageHeader` component. Both changes were committed at 453fb2f39d5596638e229743baf521b8f7c805cc (let me know if any questions or feedback on it).